python_modules/dagster/dagster/_core/definitions/time_window_partitions.py

```diff
@@ -1139,26 +1163,32 @@ def get_partition_keys_not_in_subset(
         return partition_keys
 
     def get_partition_keys(self, current_time: Optional[datetime] = None) -> Iterable[str]:
-        return [
-            pk
-            for time_window in self._included_time_windows
-            for pk in self._partitions_def.get_partition_keys_in_time_window(time_window)
-        ]
+        if self._included_partition_keys is None:
+            return [
+                pk
+                for time_window in self.included_time_windows
+                for pk in self._partitions_def.get_partition_keys_in_time_window(time_window)
+            ]
+        return sorted(
+            list(self._included_partition_keys),
+            key=lambda pkey: self._partitions_def.start_time_for_partition_key(pkey),
```

Comment:
Is this not super unperformant? Same comment from earlier applies - do we need to sort?

Reply:
ah I misread your original comment -- we have tests that rely on this being in sorted order, but I can't see how a subset would require this. I'll remove this bit